### PR TITLE
Restructured documentation + added Table of Content

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Restructured documentation + added Table of Content

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [x] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [ ] Bump-up library or tool used for development (does not change the final image)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,46 @@
...
+        * [Swagger UI](#swagger-ui-1)
+    * [Deploying OLS on OpenShift](#deploying-ols-on-openshift)
+* [Project structure](#project-structure)
+    * [Overall architecture](#overall-architecture)
+    * [Sequence diagram](#sequence-diagram)
+    * [Token truncation algorithm](#token-truncation-algorithm)
+* [Contributing](#contributing)
+* [License](#license)
+
+<!-- vim-markdown-toc -->
```
Comment (raptorsun) [inline README.md:43]: It seems these comments related to `vim-markdown-toc` are not neccessary.

Comment (tisnik) [reply README.md:43]: it was added by me. If these comments are there, the ToC will be automatically regenerated in the Editor. It's based on https://github.com/mzlogin/vim-markdown-toc but these comments seems to be compatible with other plugins for other IDEs as well.

Comment (tisnik) [reply README.md:45]: added comments about comments @raptorsun

**Comment by tisnik** [comment]
/override "Red Hat Konflux / ols-e2e-tests-rosa / lightspeed-service"
/override "Red Hat Konflux / ols-enterprise-contract / lightspeed-service"

**Pull request merged**